x/executionlayer/handler.go

```diff
@@ -97,48 +97,55 @@ func BeginBlocker(ctx sdk.Context, req abci.RequestBeginBlock, elk ExecutionLaye
 	preHash := req.Header.LastBlockId.Hash
 	unitHash := elk.GetUnitHashMap(ctx, preHash)
 
-	elk.SetCandidateBlockHash(ctx, req.Hash)
-	elk.SetUnitHashMap(ctx, req.Hash, unitHash)
+	candidateBlock := ctx.CandidateBlock()
+	candidateBlock.Hash = req.GetHash()
+	candidateBlock.State = unitHash.EEState
+
+	ctx = ctx.WithCandidateBlock(candidateBlock)
```

Comment:
Because it treats candidateBlock as a pointer, it is not necessary to call WithCandidateBlock. (2)

Reply:
https://github.com/hdac-io/friday/pull/58/commits/6f1742302db2b5a043fd34ef8f04767522701f87